AUTHORING_GUIDE.md

```diff
@@ -281,216 +188,212 @@ Args:
-
-    MESSAGES.append(payload)
-
-    # Returning any 2xx status indicates successful receipt of the message.
-    return 'OK', 200
-```
+of constraints, for example `A base64-encoded string` or `Must be between 0
+and 10`.
 
-Note the name of the function matches the URL route. The docstring is kept
-simple because reading the function body reveals how the parameters are
-used.
+### README File
 
-Use `print` or `logging.info` in request handlers to print useful information
-as needed.
+Each sample should have a `README.md` file that provides instructions for how
+to install, configure, and run the sample. Setup steps that cover creating
+Google Cloud projects and resources should link to appropriate pages in the
+[Google Cloud Documentation](https://cloud.google.com/docs/), to avoid
+duplication and simplify maintenance.
 
-### Argparse section
+### Dependencies
 
-For command-line samples, you'll need an argparse section to handle
-parsing command-line arguments and executing the sample functions. This
-section lives within the `if __name__ == '__main__'` clause:
+Every sample should include a
+[requirements.txt](https://pip.pypa.io/en/stable/user_guide/#requirements-files)
+file that lists all of its dependencies, to enable others to re-create the
+environment that was used to create and test the sample. All dependencies
+should be pinned to a specific version, as in this example:
 
-```python
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
 ```
-
-Note the use of `__doc__` (the module-level docstring) as the description for
-the parser. This helps you not repeat yourself and gives users useful
-information when invoking the program. We also use `RawDescriptionHelpFormatter`
-to prevent argparse from re-formatting the docstring.
-
-Command-line arguments should generally have a 1-to-1 match to function
-arguments. For example:
-
-```python
-parser.add_argument('source_file_name')
-parser.add_argument('destination_blob_name')
+Flask==1.1.1
+PyMySQL==0.9.3
+SQLAlchemy==1.3.12
 ```
 
-Again, descriptive names prevent you from having to exhaustively describe
-every parameter.
+If a sample has testing requirements that differ from its runtime requirements
+(such as dependencies on [pytest](http://pytest.org/en/latest/) or other
+testing libraries), the testing requirements may be listed in a separate
+`requirements-testing.txt` file instead of the main `requirements.txt` file.
 
-Some samples demonstrate multiple functions. You should use *subparsers* to
-handle this, for example:
+### Region Tags
 
-```python
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
-    parser.add_argument('bucket_name', help='Your cloud storage bucket.')
+Sample code may be integrated into Google Cloud Documentation through the use
```

Comment:
May not be true across the board, but seems like our tracking tools suggest if the samples aren't in docs that we should be deleting them? 

Love to hear what other folks are suggesting.